### PR TITLE
[Localization] Attack missed text is now localizable and uses the Gen 9 format instead of Gen 4 like before

### PR DESCRIPTION
**Pull request by CodeTappert**
## What are the changes?

It now uses the Gen 9 Format of "attackMissed". (Showing which pokemon was missed). Also localizable

Line from the orignal games:
https://abcboy101.github.io/poke-corpus/#id=sv.btl_set.BTL_STRID_SET_WazaAvoid

## Why am I doing these changes?

Because we in the wiki discord agreed that localizing the Gen 4 Message was not what we wanted.

## What did change?

See Video

### Screenshots/Videos

Before:

https://github.com/pagefaultgames/pokerogue/assets/38758606/df45c8a2-a811-405e-969c-f846779273c0

After: 

https://github.com/pagefaultgames/pokerogue/assets/38758606/c5222d9e-15d2-41ed-8906-6b3be3a22dad

## How to test the changes?

Use a KO-Move for the single target. 
For Multi Target set accuracy for earth quake for example to 1

## Checklist
- [x] There is no overlap with another PR?
  - Only with https://github.com/pagefaultgames/pokerogue/pull/2806 which was closed 
- [x] The PR is self-contained and cannot be split into smaller PRs?
  - Technically yes. But its pointless to split this up in 2 (making it localizable and the localization itself)
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/pt_BR/battle.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultNoEffect": "Isso não afeta {{pokemonName}}!",
   "hitResultOneHitKO": "Foi um nocaute de um golpe!",
   "attackFailed": "Mas falhou!",
+  "attackMissed": "{{pokemonNameWithAffix}} avoided the attack!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:28]:
```suggestion
  "attackMissed": "{{pokemonNameWithAffix}} desviou do ataque!",
```

File: src/locales/ko/battle.ts [hunk 2]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultNoEffect": "{{pokemonName}}에게는\n효과가 없는 것 같다…",
   "hitResultOneHitKO": "일격필살!",
   "attackFailed": "그러나 실패하고 말았다!!",
+  "attackMissed": "{{pokemonNameWithAffix}} avoided the attack!",
```
Comment (EnochG1) [inline src/locales/ko/battle.ts:28]:
```suggestion
  "attackMissed": "{{pokemonNameWithAffix}}에게는\n맞지 않았다!",
```

File: src/locales/fr/battle.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultNoEffect": "Ça n’affecte pas {{pokemonName}}…",
   "hitResultOneHitKO": "K.O. en un coup !",
   "attackFailed": "Mais cela échoue !",
+  "attackMissed": "{{pokemonNameWithAffix}} avoided the attack!",
```
Comment (Adri1) [inline src/locales/fr/battle.ts:28]:
```suggestion
  "attackMissed": "{{pokemonNameWithAffix}}\névite l’attaque !",
```

**Review by Adri1 — COMMENTED** [review]
Added french

File: src/locales/zh_CN/battle.ts [hunk 4]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultNoEffect": "对{{pokemonName}}没有效果！！",
   "hitResultOneHitKO": "一击必杀！",
   "attackFailed": "但是失败了！",
+  "attackMissed": "{{pokemonNameWithAffix}} avoided the attack!",
```
Comment (40chyan) [inline src/locales/zh_CN/battle.ts:28]:
```suggestion
  "attackMissed": "没有命中{{pokemonNameWithAffix}}！",
```

File: src/locales/zh_TW/battle.ts [hunk 5]
```diff
@@ -22,6 +22,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultNoEffect": "對 {{pokemonName}} 沒有效果！",
   "hitResultOneHitKO": "一擊切殺！",
   "attackFailed": "但是失敗了！",
+  "attackMissed": "{{pokemonNameWithAffix}} avoided the attack!",
```
Comment (40chyan) [inline src/locales/zh_TW/battle.ts:25]:
```suggestion
  "attackMissed": "沒有命中{{pokemonNameWithAffix}}！",
```

**Comment by 40chyan** [comment]
**Also a critical reason to change this is because we don't have gen4 official translation in zh_cn and zh_tw**

File: src/locales/it/battle.ts [hunk 6]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultNoEffect": "Non ha effetto su {{pokemonName}}!",
   "hitResultOneHitKO": "KO con un colpo!",
   "attackFailed": "Ma ha fallito!",
+  "attackMissed": "{{pokemonNameWithAffix}} avoided the attack!",
```
Comment (Adri1) [inline src/locales/it/battle.ts:28]:
```suggestion
  "attackMissed": "{{pokemonNameWithAffix}} evita l’attacco!",
```

File: src/locales/es/battle.ts [hunk 7]
```diff
@@ -25,6 +25,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultNoEffect": "No afecta a {{pokemonName}}!",
   "hitResultOneHitKO": "¡KO en 1 golpe!",
   "attackFailed": "¡Pero ha fallado!",
+  "attackMissed": "{{pokemonNameWithAffix}} avoided the attack!",
```
Comment (Adri1) [inline src/locales/es/battle.ts:28]:
```suggestion
  "attackMissed": "¡{{pokemonNameWithAffix}} ha evitado el ataque!",
```

**Review by Adri1 — COMMENTED** [review]
Added Spanish and Italian according to Scarlet and Violet texts
image

**Review by returntoice — APPROVED** [review]

File: src/phases.ts [hunk 8]
```diff
@@ -2894,7 +2894,10 @@ export class MoveEffectPhase extends PokemonPhase {
       if (!activeTargets.length || (!move.hasAttr(VariableTargetAttr) && !move.isMultiTarget() && !targetHitChecks[this.targets[0]])) {
         this.stopMultiHit();
         if (activeTargets.length) {
-          this.scene.queueMessage(getPokemonMessage(user, "'s\nattack missed!"));
+
+          this.scene.queueMessage(i18next.t("battle:attackMissed", { pokemonNameWithAffix: getPokemonNameWithAffix(this.getTarget()) }));
+
+
```
Comment (CodeTappert) [inline src/phases.ts:2900]:
```suggestion
          this.scene.queueMessage(i18next.t("battle:attackMissed", { pokemonNameWithAffix: getPokemonNameWithAffix(this.getTarget()) }));
```

**Review by Adri1 — APPROVED** [review]

**Review by flx-sta — APPROVED** [review]

**Review by f-fsantos — APPROVED** [review]

**Pull request merged**